pkg/fqdn/rulegen.go

```diff
@@ -103,15 +103,21 @@ func NewRuleGen(config Config) *RuleGen {
 
 }
 
-// MarkToFQDNRules adds a tracking label to rules that contain ToFQDN sections.
+// PrepareFQDNRules adds a tracking label to rules that contain ToFQDN sections.
 // The label is used to ensure that the ToFQDN rules are replaced correctly
 // when they are regenerated with IPs. It will also include the generated IPs
 // (in the ToCIDRSet) section for DNS names already present in the cache.
+// It returns the list of valid rules to apply into policyRepo, a valid rule is:
+// - A rule without toFQDN entries.
+// - A rule with toFQDN that already have a UUID and it is present on
+// gen.AllRules
+// - A new FQDN rule that does not have UUID(new UUID will be created in this function)
 // NOTE: It edits the rules in-place
-func (gen *RuleGen) MarkToFQDNRules(sourceRules []*api.Rule) {
+func (gen *RuleGen) PrepareFQDNRules(sourceRules []*api.Rule) []*api.Rule {
 	gen.Lock()
 	defer gen.Unlock()
 
+	result := []*api.Rule{}
```

Comment:
Do we need to worry about the difference between a nil slice and non-nil empty? I think no but this forces it one way.

Reply:
no, we don't need to worry about!